cocotb/types/__init__.py

```diff
@@ -3,3 +3,28 @@
 # SPDX-License-Identifier: BSD-3-Clause
 from .range import Range  # noqa: F401
 from .array import Array  # noqa: F401
+
+
+def concat(a: Array, b: Array) -> Array:
+    """
+    Create a new array that is the concatenation of one array with another.
+
+    Uses the :meth:`__concat__` or :meth:`__rconcat__` special methods to dispatch to a particular implementation,
+    exactly like other binary operations in Python.
+
+    Raises:
+        TypeError: when *other* is an object of dissimilar type.
```

Comment:
```suggestion
        TypeError: when *b* is an object of dissimilar type.
```

Reply:
I think it needs a rewrite.
```suggestion
        TypeError: when concatenation is not supported between the two object.
```